### PR TITLE
feat: calculate order

### PR DESCRIPTION
**Pull request by BruceClark406**

### CODE REVIEW

File: lib/src/square_api_client.dart [hunk 1]
```diff
@@ -103,6 +103,10 @@ abstract class SquareApiClient {
     @Path() String orderId,
     @Body() PayOrderRequest payOrderRequest,
   );
+  @POST('/v2/orders/calculate')
+  Future<PayOrderResponse> calculateOrder(
```
Comment (mtwichel) [inline lib/src/square_api_client.dart:107]:
```suggestion
  Future<CalculateOrderResponse> calculateOrder(
```

File: lib/src/functions_model/calculate_order.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  @override
+  List<Object?> get props => [
+        order,
+        orderReward,
+      ];
+}
+
+@JsonSerializable()
+class CalculateOrderResponse extends SquareResponse with EquatableMixin {
```
Comment (mtwichel) [inline lib/src/functions_model/calculate_order.dart:33]:
```suggestion
class CalculateOrderResponse extends SquareResponse {
```

File: lib/src/functions_model/calculate_order.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
```
Comment (mtwichel) [inline lib/src/functions_model/calculate_order.dart:3]:
```suggestion
```

File: lib/src/functions_model/calculate_order.dart [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import 'package:square_connect/square_connect.dart';
+
+part 'calculate_order.g.dart';
+
+@JsonSerializable()
+class CalculateOrderRequest extends Equatable {
+  const CalculateOrderRequest({
+    required this.order,
+    required this.orderReward,
```
Comment (mtwichel) [inline lib/src/functions_model/calculate_order.dart:12]:
```suggestion
   this.proposedRewards ,
```

File: lib/src/functions_model/calculate_order.dart [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  /// Converts a [Map] to an [CalculateOrderRequest]
+  factory CalculateOrderRequest.fromJson(Map<String, dynamic> json) =>
+      _$CalculateOrderRequestFromJson(json);
+
+  /// Converts a [CalculateOrderRequest] to a [Map]
+  Map<String, dynamic> toJson() => _$CalculateOrderRequestToJson(this);
+
+  final Order order;
+  final List<OrderReward>? orderReward;
```
Comment (mtwichel) [inline lib/src/functions_model/calculate_order.dart:23]:
```suggestion
  final List<OrderReward>? proposedRewards ;
```

**Review by mtwichel — CHANGES_REQUESTED** [review]

File: lib/src/functions_model/calculate_order.dart (hunk 2 above)
Comment (BruceClark406) [reply lib/src/functions_model/calculate_order.dart:33]: Cool!

**Review by mtwichel — APPROVED** [review]
LGTM! Thanks for the contribution!

**Pull request merged**